gcp/biganimal_GCP_baserole.yaml

```diff
@@ -0,0 +1,540 @@
+- container.runtimeClasses.delete
+- container.runtimeClasses.get
+- container.runtimeClasses.list
+- container.runtimeClasses.update
+- container.secrets.create
+- container.secrets.delete
+- container.secrets.get
+- container.secrets.list
+- container.secrets.update
+- container.selfSubjectAccessReviews.create
+- container.selfSubjectRulesReviews.create
+- container.serviceAccounts.create
+- container.serviceAccounts.createToken
+- container.serviceAccounts.delete
+- container.serviceAccounts.get
+- container.serviceAccounts.list
+- container.serviceAccounts.update
+- container.services.create
+- container.services.delete
+- container.services.get
+- container.services.getStatus
+- container.services.list
+- container.services.proxy
+- container.services.update
+- container.services.updateStatus
+- container.statefulSets.create
+- container.statefulSets.delete
+- container.statefulSets.get
+- container.statefulSets.getScale
+- container.statefulSets.getStatus
+- container.statefulSets.list
+- container.statefulSets.update
+- container.statefulSets.updateScale
+- container.statefulSets.updateStatus
+- container.storageClasses.create
+- container.storageClasses.delete
+- container.storageClasses.get
+- container.storageClasses.list
+- container.storageClasses.update
+- container.subjectAccessReviews.create
+- container.thirdPartyObjects.create
+- container.thirdPartyObjects.delete
+- container.thirdPartyObjects.get
+- container.thirdPartyObjects.list
+- container.thirdPartyObjects.update
+- container.tokenReviews.create
+- container.updateInfos.create
+- container.updateInfos.delete
+- container.updateInfos.get
+- container.updateInfos.list
+- container.updateInfos.update
+- container.validatingWebhookConfigurations.create
+- container.validatingWebhookConfigurations.delete
+- container.validatingWebhookConfigurations.get
+- container.validatingWebhookConfigurations.list
+- container.validatingWebhookConfigurations.update
+- container.volumeAttachments.create
+- container.volumeAttachments.delete
+- container.volumeAttachments.get
+- container.volumeAttachments.getStatus
+- container.volumeAttachments.list
+- container.volumeAttachments.update
+- container.volumeAttachments.updateStatus
+- container.volumeSnapshotClasses.create
+- container.volumeSnapshotClasses.delete
+- container.volumeSnapshotClasses.get
+- container.volumeSnapshotClasses.list
+- container.volumeSnapshotClasses.update
+- container.volumeSnapshotContents.create
+- container.volumeSnapshotContents.delete
+- container.volumeSnapshotContents.get
+- container.volumeSnapshotContents.getStatus
+- container.volumeSnapshotContents.list
+- container.volumeSnapshotContents.update
+- container.volumeSnapshotContents.updateStatus
+- container.volumeSnapshots.create
+- container.volumeSnapshots.delete
+- container.volumeSnapshots.get
+- container.volumeSnapshots.getStatus
+- container.volumeSnapshots.list
+- container.volumeSnapshots.update
+- container.volumeSnapshots.updateStatus
```

Comment:
What's the effect of GCP level GKE permissions? Is this handled as in AZ's (AAD RBAC integration for k8s)[https://learn.microsoft.com/en-us/azure/aks/concepts-identity#azure-rbac-for-kubernetes-authorization]? In particular are these reflected inside clusters with the creation of roles & bindings? Or are they checked with a webhook as in AZ?

Are these needed at GCP level so that there is a first cluster admin at cluster bootstrap, before any role/binding is created?

I'm asking because I find it a bit problematic that one can't get a picture of what identities can do what on a cluster by just looking at the roles and binding in a cluster. I realize the pattern above is probably unavoidable, but I think it should be kept in check.

Reply:
The easy answer to why these are required is that the [Terraform module that we use to create GKE clusters](https://registry.terraform.io/modules/terraform-google-modules/kubernetes-engine/google) has the `roles/container.developer` GCP IAM built-in role as a prerequisite, and it has much of these rights assigned to it, as we can see [here](https://cloud.google.com/iam/docs/understanding-roles#container.developer). Then, after the GKE cluster creation, our Terraform modules are also installing and configuring a few Kubernetes components in the cluster, and these surely requires access to a subset of these rights as well.